src/main/java/de/tum/cit/aet/artemis/atlas/service/competency/StandardizedCompetencyService.java

```diff
@@ -191,6 +191,7 @@ public void importStandardizedCompetencyCatalog(StandardizedCompetencyCatalogDTO
      */
     public String exportStandardizedCompetencyCatalog() {
         List<KnowledgeArea> knowledgeAreas = getAllForTreeView();
+        // TODO: we should avoid using findAll() here, as it might return a huge amount of data
```

Comment:
I do not see how this is a valid todo.

When exporting the comptency catalog of standardized competencies, users expect the catalog with all the sources it is composed of. Thus, this is one of the rare cases where a simple `findAll()` is reasonable.

Reply:
I think we should generally forbid the use of findAll. I would like to enforce this at some point with ArchUnit tests.

You never know how much the data grows at some point. What happens if other subject areas next to computer science provide standardized catalogs as well? 

A user typically cannot read more than 50-100 entries on one page. Paging (potentially combined with continuous scrolling and fetching) can be used in such cases to avoid findAll.

I'm also fine setting the page size to 200 in this specific case and only implement paging when we are sure there are more entries